### DISCUSSION TITLE
v3.1.4 - minor improvement

### DISCUSSION THREAD

**Discussion by alexdelprete** [alexdelprete/ha-abb-powerone-pvi-sunspec#84] [Announcements]
- rewrote **check_port()**

<hr /><em>This discussion was created from the release <a href='https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/releases/tag/v3.1.4'>v3.1.4 - minor improvement</a>.</em>

**Comment by lucasdaly90** [comment]
Hey buddy,

Thanks for this project - I'm making some progress!

Saw these in logs and thought I should report it to you.

(opt_comm_model) Model unknown, report to @alexdelprete on the forum the following data: Manuf.: B-3 - Model: P25-0x - Options: 04ED/0xFF4B/22 - OptModel: 04ED/0xFF4B/22 - OptModelInt: 48
(opt_comm_model) Model unknown, report to @alexdelprete on the forum the following data: Manuf.: ABB - Model: -3P25- - Options: 0x04ED/0xFF4B/ - OptModel: 0x04ED/0xFF4B/ - OptModelInt: 4

I've got an ABB UNO-DM-5.0-TL-PLUS-B Inverter.

**Comment by alexdelprete** [comment]
Hi Lucas,

I don't understand the logs: you have 2 inverters? I see different data...the first line of log reported looks wrong...like shifted of a couple of bytes...:)

The OptModelInt: 4 of the second log seems correct, it's missing in the list, and it's among its "brothers":

```
DEVICE_MODEL = {
      0: "UNO-DM-3.3-TL-PLUS",
      1: "UNO-DM-4.0-TL-PLUS",
      3: "UNO-DM-4.6-TL-PLUS",
      5: "UNO-DM-6.0-TL-PLUS",
     11: "UNO-DM-2.0-TL-PLUS",
```

I'll include it and in next release you'll see the correct model string in the sensor.

**Comment by lucasdaly90** [comment]
I only have a single inverter. 
Now that you mention it, it does look like all the characters have shifted position by 2 on the first line. Bizarre. 

Thanks for including it. 
Peace

**Reply by alexdelprete** [reply]
Were you using a different configuration? Different starting address?

**Reply by lucasdaly90** [reply]
I did try a few starting addresses before I struck gold on 40000

**Reply by alexdelprete** [reply]
Got it. Stick with the docs. :)

**Reply by lucasdaly90** [reply]
Yep. I should default to reading the README. But where’s the fun in that?

**Reply by alexdelprete** [reply]
Perfectly agree!! :D
